picard/ui/options/general.py

```diff
@@ -158,17 +161,36 @@ def update_login_logout(self, error_msg=None):
     def login(self):
         self.tagger.mb_login(self.on_login_finished, self)
 
-    def restore_defaults(self):
-        super().restore_defaults()
-        self.logout()
-
     def on_login_finished(self, successful, error_msg=None):
         self.update_login_logout(error_msg)
 
     def logout(self):
-        self.tagger.mb_logout()
+        self.tagger.mb_logout(self.on_logout_finished)
+
+    def on_logout_finished(self, successful, error_msg=None):
+        if not successful:
+            msg = QtWidgets.QMessageBox(self)
+            msg.setIcon(QtWidgets.QMessageBox.Icon.Warning)
+            msg.setWindowTitle(_("Logout error"))
+            msg.setText(_(
+                "A server error occurred while revoking access to the MusicBrainz server: %s\n\nRemove locally stored credentials anyway?") % error_msg)
```

Comment:
Just a minor thing, since some time I started to split those long strings with newlines at newlines:

```python
         msg.setText(_(
                "A server error occurred while revoking access to the MusicBrainz server: %s\n"
                "\n"
                "Remove locally stored credentials anyway?") % error_msg)
```

It is easier to read as it is more or less formatted as it will be written, and it doesn't change anything to the actual string

Reply:
That makes sense, looks good. I'll update this PR prior to merging.

Reply:
Changed as suggested